front/src/common/IntervalsEditor/IntervalsEditorCommonForm.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
+import { fixLinearMetadataItems, resizeSegment } from 'common/IntervalsDataViz/data';
+import { notEmpty } from 'common/IntervalsDataViz/utils';
+import { IntervalItem } from './types';
+
+type IntervalsEditorFormProps = {
+  data: IntervalItem[];
+  interval: IntervalItem;
+  selectedIntervalIndex: number;
+  setData: (newData: IntervalItem[]) => void;
+  totalLength: number;
+};
+
+const IntervalsEditorCommonForm = ({
+  data,
+  interval,
+  selectedIntervalIndex,
+  setData,
+  totalLength,
+}: IntervalsEditorFormProps) => {
+  const { t } = useTranslation();
+
+  /**
+   * Resize segment from inputs
+   */
+  // TODO: fix bug
```

Comment:
Just to be sure, you are referring to the strange behaviour when we type a new value in the input ?

Reply:
Yes, and there was also another bug (resizing does not work after splitting a segment I think !) but everything is fixed in another PR :)